colmena/proxy.py

```diff
@@ -1,18 +1,65 @@
 """Utilities for interacting with ProxyStore"""
 import logging
 import warnings
-import proxystore as ps
 
-from typing import Any, Union, List
+import proxystore
+from proxystore.store.endpoint import EndpointStore
+from proxystore.store.file import FileStore
+from proxystore.store.globus import GlobusStore
+from proxystore.store.local import LocalStore
+from proxystore.store.redis import RedisStore
+from proxystore.proxy import extract
+from proxystore.proxy import is_resolved
+from proxystore.proxy import Proxy
+from proxystore.store.base import Store
+from proxystore.store.utils import resolve_async
+
+from typing import Any, Union, List, Optional, Type
 
 logger = logging.getLogger(__name__)
 
+STORES = {
+    'EndpointStore': EndpointStore,
+    'FileStore': FileStore,
+    'GlobusStore': GlobusStore,
+    'LocalStore': LocalStore,
+    'RedisStore': RedisStore,
+}
```

Comment:
Is there a plan to eventually put this kind of "string->Store" mapping in ProxyStore? I'm worried about the having to continually update this area as PS gets more store types.

Reply:
No, it was actually removed from ProxyStore hence this addition. 

In ProxyStore, it had two annoying aspects. 1) All `Store`s needed to be importable which prevented us from making an "optional" `Store` only available via an `extras_install` (also caused mocking/circular import troubles because a class was imported in a parent `__init__.py`. 2) Extending the `STORES` enum is not possible at runtime so lookup by string if you "bring your own `Store`" is not possible.

I wonder if a better solution is to store the full module path, e.g., `proxystore.store.file.FileStore`, and then do an `__import__` + module level `__getattr__` to dynamically resolve the correct type from a string. Seems more robust to me and would work with `Store`s in other packages. What do you think?

Reply:
Having a lookup by full path is a good idea, given that this current implementation requires all store types to be importable.